ipywidgets/static/widgets/js/manager.js

```diff
@@ -306,6 +306,22 @@ define([
 
     WidgetManager.prototype.create_model = function (options) {
         /**
+         * For backward compatibility. Custom widgets may be relying on the fact
+         * that create_model was creating a comm if none was provided in options.
+         * Unlike the old version of create_model, if no comm is passed,
+         * options.model_id is used to create the new comm.
+         */
+        console.warn('WidgetManager.create_model is deprecated. User WidgetManager.new_model');
+        if (!options.comm) {
+            options.comm = this.comm_manager.new_comm('ipython.widget',
+                                                      {'widget_class': options.widget_class},
+                                                       options.model_id);
+        }
+        return this.new_model(options);
+    }
+
+    WidgetManager.prototype.new_model = function (options) {
```

Comment:
because `create_model` creates a new comm when it is not passed one. Hence it does not allow comm-less models.